### PR TITLE
(go) Fix escaped character literals

### LINKED ISSUE #1411: (go) Misunderstands Backslash Character Literals
In Go, when invoking the escape sequence for backslash in a character literal, HLJS gets stuck incorrectly interpreting the following lines as part of a string. Observe the following simple go program.

    package main
    import "fmt"
    func main() {
      s := "\\"
      c := '\\'
      fmt.Println("Hello, 世界", s, c)
    }

When highlighted with HLJS 9.5.0, the entire file following the `'\\'` is highlighted as a string. It doesn't have this issue with the string literal (`"\\"`).

### PR DESCRIPTION
**Pull request by davidben**
Fixes #1411

### CODE REVIEW

File: src/languages/go.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ function(hljs) {
         className: 'string',
         variants: [
           hljs.QUOTE_STRING_MODE,
-          {begin: '\'', end: '[^\\\\]\''},
```
Comment (yyyc514) [inline src/languages/go.js]: I’ve seen this a few places now. Does it just not work as intended at all?  Is it only escaping two slashes or?  Seems like it wouldn’t match super short strings either?

Comment (davidben) [reply src/languages/go.js]: Yeah, it just doesn't work. It's looking for a `'` not preceded by `\`, but `'\\'` is perfectly legitimate and fails the regex.

**Comment by yyyc514** [comment]
I’ll review all these PRs more carefully tomorrow.

**Comment by yyyc514** [comment]
Any chance you have time/interest to look into the 6-7 other languages we're using this and see if they likewise suffer from the same issue?

One could imagine a language where \\ wasn't valid so this MIGHT be a reasonable regex...

File: src/languages/go.js (hunk 1 above)
Comment (yyyc514) [reply src/languages/go.js]: \Yeah I get confused by all the extra escaping with '' strings.  Looks like exactly what you say.

**Comment by davidben** [comment]
> Any chance you have time/interest to look into the 6-7 other languages we're using this and see if they likewise suffer from the same issue?

I'm traveling for the next two weeks so probably not for a bit. But maybe if I get bored on the plane.

> Please add Changelog entry.

Done

**Pull request merged**